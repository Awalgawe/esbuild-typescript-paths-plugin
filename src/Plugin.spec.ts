import { PluginBuild, build } from 'esbuild';
import { resolve } from 'node:path';
import TsPathsPlugin from './Plugin';
import { describe, it, expect } from '@jest/globals';

describe('Plugin', () => {
  it('should work', async () => {
    const plugin = new TsPathsPlugin();

    // @ts-expect-error Mocking esbuild
    const esbuild: PluginBuild = {
      onLoad: () => {},
      onResolve: () => {},
      initialOptions: {
        absWorkingDir: resolve(process.cwd(), 'assets'),
        tsconfig: './tsconfig.json',
      },
    };

    await plugin.setup(esbuild);

    const result = await plugin.onLoad({
      namespace: 'file',
      suffix: 'js',
      path: resolve(process.cwd(), 'assets/src/test-imports.ts'),
      pluginData: {},
    });

    expect(result?.contents).toMatchSnapshot('yolo');
  });

  it('build bundle', async () => {
    const { name, setup } = new TsPathsPlugin();

    const result = await build({
      absWorkingDir: resolve(process.cwd(), 'assets'),
      tsconfig: './tsconfig.json',
      entryPoints: [resolve(process.cwd(), 'assets/src/a')],
      plugins: [{ name, setup }],
      target: 'es2022',
      format: 'esm',
      logLevel: 'info',
      outdir: 'toto',
      bundle: true,
      write: false,
    });

    expect(result.errors).toStrictEqual([]);
  });

  it('build without bundling', async () => {
    const { name, setup } = new TsPathsPlugin();

    const result = await build({
      absWorkingDir: resolve(process.cwd(), 'assets'),
      tsconfig: './tsconfig.json',
      entryPoints: [resolve(process.cwd(), 'assets/src/a/**/*')],
      plugins: [{ name, setup }],
      target: 'es2022',
      format: 'esm',
      logLevel: 'info',
      outdir: 'toto',
      bundle: false,
      write: false,
    });

    expect(result.errors).toStrictEqual([]);
  });
});
