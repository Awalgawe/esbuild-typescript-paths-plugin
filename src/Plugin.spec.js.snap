// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin should work: yolo 1`] = `
"/* eslint-disable prettier/prettier */
/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-var-requires */
// @ts-nocheck

// ES6 Import
import module from './1/module';

// ES6 Named Import
import { namedExport } from './2/named-module';

// ES6 Import with Alias
import * as alias from './3/alias-module';

// ES6 Import Default and Named
import defaultExport, {
  namedExport as renamedExport,
} from './4/multi-module';

// ES6 Import Default and Alias
import defaultExport, * as alias from './5/alias-multi-module';

// CommonJS Require
const commonjsModule = require('./6/commonjs-module');

// CommonJS Named Require
const { commonjsNamedExport } = require('./7/commonjs-named-module');

// Dynamic Import (ES6)
const dynamicModule = await import('./8/dynamic-module');

// Side Effect Import (ES6)
import './9/side-effect-module';

// Side Effect Require (CommonJS)
require('./10/side-effect-commonjs-module');

// ES6 Default Export
export default {
  key: 'value',
};

// ES6 Named Export
export const namedValue1 = 'Hello';

// ES6 Renamed Export
export { namedValue1 as renamedValue };

// ES6 Export List
export { value1, value2 } from './11/export-list-module';

// ES6 Re-export from Another Module
export { namedValue2 } from './12/reexport-module';

// ES6 All Named Exports
export * from './13/all-exports-module';

// External Module
import { namedValue3 } from '../packages/14/external-module';
"
`;
